src/node-file-trace.js

```diff
@@ -224,6 +228,14 @@ class Job {
     return true;
   }
 
+  getPjsonBoundary (path) {
+    const root = path.slice(0, path.indexOf(sep)) || path.sep;
```

Comment:
I would expect `root` to always be `path.sep`. It seems like this implementation will skip the system root directory.

Also, should this account for the `base` option? Maybe that should be considered the `root`?

Reply:
This is the boundary algorithm from https://nodejs.org/dist/latest-v14.x/docs/api/esm.html#esm_resolver_algorithm READ_PACKAGE_SCOPE.

The root should be `C:\\` on windows and `/` on posix. If I got the logic wrong this might be the cause of the loop. What is weird is I'm testing on WSL Ubuntu here so would have thought I'd seen that.

Reply:
Ah this likely was the cause - I wasn't running the ecmascript modules tests locally which exactly hit this root scenario. Test should be passing now.